api/src/test/java/com/ford/labs/retroquest/deprecated_tests/BoardServiceTest.java

```diff
@@ -103,21 +108,50 @@ void getBoardsForTeamId_shouldReturnAPagedResult() {
 
     @Test
     void saveBoard() {
-        Board boardToSave = Board.builder()
-                .teamId("team1")
-                .thoughts(Collections.singletonList(Thought.builder().message("hello").build()))
-                .build();
-
-        Board savedBoard = Board.builder()
-                .id(1L)
-                .teamId("team1")
-                .dateCreated(LocalDate.now())
-                .thoughts(Collections.singletonList(Thought.builder().message("hello").build()))
-                .build();
-
-        when(boardRepository.save(boardToSave)).thenReturn(savedBoard);
+        var expectedTeamId = "team1";
+        var expectedMessage = "hello";
+        var boardToSave = new CreateBoardRequest(
+            expectedTeamId,
+            List.of(
+                new CreateThoughtRequest(
+                    expectedMessage,
+                    0,
+                    null,
+                    false,
+                    null,
+                    null
+                )
+            )
+        );
 
-        Board returnedBoard = boardService.saveBoard(boardToSave);
-        assertEquals(savedBoard, returnedBoard);
+        when(boardRepository.save(any(Board.class))).thenAnswer(a -> {
+            var board = a.<Board>getArgument(0);
+            board.setId(1234L);
+            return board;
+        });
+        when(thoughtService.createThought(anyString(), anyLong(), any(CreateThoughtRequest.class)))
+            .thenAnswer(a -> {
+                var teamId = a.<String>getArgument(0);
+                var boardId = a.<Long>getArgument(1);
+                var request = a.<CreateThoughtRequest>getArgument(2);
+                return new Thought(
+                    4321L,
+                    request.getMessage(),
+                    request.getHearts(),
+                    request.getTopic(),
+                    request.isDiscussed(),
+                    teamId,
+                    null,
+                    boardId
+                );
+            });
+
+        var returnedBoard = boardService.createBoard(boardToSave);
+
+        assertThat(returnedBoard.getId()).isEqualTo(1234L);
+        assertThat(returnedBoard.getTeamId()).isEqualTo(expectedTeamId);
+        assertThat(returnedBoard.getThoughts()).hasSize(1);
+        assertThat(returnedBoard.getThoughts()).hasSize(1);
```

Comment:
Why do we check the size of the thoughts twice?

Reply:
Typo. I've fixed it.